test/types/schema.test.ts

```diff
@@ -429,7 +429,7 @@ export function autoTypedSchema() {
     mixed3: Schema.Types.Mixed,
     objectId1: Schema.Types.ObjectId,
     objectId2: 'ObjectId',
-    objectId3: 'ObjectID',
+    objectId3: 'objectId',
```

Comment:
@vkarpov15  
It is a bit confusing about the supported strings that should be resolved to ```Types.ObjectId```,
Please clarify which ones that are supported.

Reply:
@iammola 
Please add this test to the end of models.test.ts
```ts
function gh12168() {
  const TermSchema = new Schema({
    session: { type: Schema.Types.ObjectId, required: true, ref: '...' },
    start: { type: Date, required: true, unique: true },
    end: { type: Date, required: true, unique: true }
  });

  const TermModel = model('test', TermSchema);

  expectType<{ session: Types.ObjectId; start: Date; end: Date; }>({} as InferSchemaType<typeof TermSchema>);
}
```

Reply:
@mohammad0-0ahmad I took a look and it doesn't look like we support 'objectid', the below JavaScript throws an error:

```
'use strict';

const mongoose = require('mongoose');

const schema = new mongoose.Schema({
  fooId: 'objectid' // Throws "TypeError: Invalid schema configuration: `Objectid` is not a valid type at path `fooId`."
});

console.log('Done');
```

So the 'objectid' change is unnecessary and incorrect

And I would rather avoid reverting the auto typed schema feature. It does make TypeScript a little slower, but on the other hand it also removes the need for document interfaces. We'll figure out some other way to make things faster. I've learned that there's always something you can do to come up with functionally equivalent types that don't cause tsc to grind to a halt, it's just a guessing game to figure out what that is :)

Reply:
Thanks, @vkarpov15, That makes me feel better. Without doubt, we will find a solution for that :)